consensus/validator.go

```diff
@@ -325,6 +293,68 @@ func (consensus *Consensus) onCommitted(msg *msg_pb.Message) {
 	consensus.consensusTimeout[timeoutConsensus].Start()
 }
 
+// Collect private keys that are part of the current committee.
+// TODO: cache valid private keys and only update when keys change.
+func (consensus *Consensus) getPriKeysInCommittee() []*bls.PrivateKeyWrapper {
+	priKeys := []*bls.PrivateKeyWrapper{}
+	for i, key := range consensus.priKey {
+		if !consensus.IsValidatorInCommittee(key.Pub.Bytes) {
+			continue
+		}
+		priKeys = append(priKeys, &consensus.priKey[i])
+	}
+	return priKeys
+}
+
+func (consensus *Consensus) constructP2pMessages(msgType msg_pb.MessageType, payloadForSign []byte, priKeys []*bls.PrivateKeyWrapper) []*NetworkMessage {
+	p2pMsgs := []*NetworkMessage{}
+	if consensus.AggregateSig {
+		networkMessage, err := consensus.construct(msgType, payloadForSign, priKeys)
+		if err != nil {
+			logger := consensus.getLogger().Err(err).
+				Str("message-type", msgType.String())
+			for _, key := range priKeys {
+				logger.Str("key", key.Pri.SerializeToHexStr())
+			}
+			logger.Msg("could not construct message")
+		} else {
+			p2pMsgs = append(p2pMsgs, networkMessage)
+		}
+
+	} else {
+		for _, key := range priKeys {
+			networkMessage, err := consensus.construct(msgType, payloadForSign, []*bls.PrivateKeyWrapper{key})
+			if err != nil {
+				consensus.getLogger().Err(err).
```

Comment:
redundant getLogger here, just reuse `logger`

Reply:
not really redundant. the logger was initialized for the first if condition. This else {} doesn't have it.